tests/beaker_tests/cisco_itd_service/test_itd_service.rb

```diff
@@ -0,0 +1,215 @@
+###############################################################################
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# Test hash top-level keys
+tests = {
+  master:        master,
+  agent:         agent,
+  platform:      'n(7|9)k',
+  resource_name: 'cisco_itd_service',
+}
+
+# Test hash test cases
+tests[:default] = {
+  desc:           '1.1 Defaults',
+  title_pattern:  'myService',
+  preclean:       'cisco_itd_service',
+  manifest_props: {
+    access_list:         'default',
+    device_group:        'default',
+    exclude_access_list: 'default',
+    fail_action:         'default',
+    ingress_interface:   'default',
+    load_bal_enable:     'default',
+    shutdown:            'default',
+    virtual_ip:          'default',
+  },
+  code:           [0, 2],
+  resource:       {
+    fail_action:     'false',
+    load_bal_enable: 'false',
+    shutdown:        'true',
+  },
+}
+
+tests[:default_plat_1] = {
+  desc:           '1.2 Defaults for platform specific part 1',
```

Comment:
If I understand this correctly, you have a default test for n7k and then one for n9k right?  If so, just set the `desk:` to `1.2 Defaults for n9k platform`, and then do the same for `n7k`.